DSA_Session5/src/dictionary/DictionaryImp.java

```diff
@@ -0,0 +1,273 @@
+package dictionary;
+
+import java.util.*;
+
+import org.json.simple.JSONObject;
+
+/**
+ *Class DictionaryImp that implements dictionary interface
+ *
+ *@author Aman Gautam
+ * 
+ * Date - 06/08/2019
+ */
+public class DictionaryImp implements Dictionary {
+	BSTNode head;
+	
+	/**
+	 * parameterized constructor
+	 * @input json object
+   	 */
+	public DictionaryImp(JSONObject jsonObject) throws Exception {
+		this.head = null;
+		this.input(jsonObject);
+	}
+	
+	//default constructor
+	public DictionaryImp() {
+	}
+
+	/**
+	 * method for add key and value from json object
+	 * @param jsonObject
+	 */
+	private boolean input(JSONObject jsonObject) throws Exception{
+		
+		for(Object obj : jsonObject.keySet()){
+			String key = (String)obj;
+			String value = (String) jsonObject.get(key);
+			this.add(key, value);
+		}
+		return true;
+	}
+
+	/**
+	 * method for add key and value in BST
+	 */
+	@Override
+	public boolean add(String key, String value) throws Exception {
```

Comment:
In string add() method case there no need to handle the exception. Because we do not have any case for occurring exception.